File: examples/access/locations/working_with_properties.py
# /// script
# dependencies = ["brahe"]
# ///
"""
Demonstrate adding and retrieving custom properties on locations.
Shows scalar, string, and boolean property types.
"""

import brahe as bh

bh.initialize_eop()

location = bh.PointLocation(-122.4194, 37.7749, 0.0)

# Add scalar properties
location.add_property("antenna_gain_db", 42.5)
location.add_property("frequency_mhz", 8450.0)

# Add string properties
location.add_property("operator", "NOAA")

# Add boolean flags
location.add_property("uplink_enabled", True)

# Retrieve properties
props = location.properties
gain = props.get("antenna_gain_db")
operator = props.get("operator")
uplink = props.get("uplink_enabled")

print(f"Antenna Gain: {gain}")
print(f"Operator: {operator}")
print(f"Uplink Enabled: {uplink}")

# Expected output:
# Antenna Gain: 42.5
# Operator: NOAA
# Uplink Enabled: True

File: examples/access/locations/working_with_properties.rs
//! Demonstrate adding and retrieving custom properties on locations.
//! Shows scalar, string, and boolean property types.

#[allow(unused_imports)]
use brahe as bh;
use bh::AccessibleLocation;
use serde_json::json;

fn main() {
    bh::initialize_eop().unwrap();

    let location = bh::PointLocation::new(-122.4194, 37.7749, 0.0)
        .add_property("antenna_gain_db", json!(42.5))
        .add_property("frequency_mhz", json!(8450.0))
        .add_property("operator", json!("NOAA"))
        .add_property("uplink_enabled", json!(true));

    // Access properties
    let props = location.properties();
    if let Some(gain) = props.get("antenna_gain_db") {
        println!("Antenna Gain: {}", gain);
    }
    if let Some(operator) = props.get("operator") {
        println!("Operator: {}", operator);
    }
    if let Some(uplink) = props.get("uplink_enabled") {
        println!("Uplink Enabled: {}", uplink);
    }

    // Expected output:
    // Antenna Gain: 42.5
    // Operator: "NOAA"
    // Uplink Enabled: true
}
